src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -347,10 +296,118 @@ HTTPProxy::_handleGetSchema(shd::GetSchemaRequest&& request) {
+        k2val = dtoexp::makeValueReference(shval.fieldName);
+    } else if (shval.type == shd::FieldType::NULL_T || shval.type == shd::FieldType::NOT_KNOWN || shval.type == shd::FieldType::NULL_LAST) {
+        k2val.type =  static_cast<dto::FieldType>(to_integral(shval.type));
+    } else {
+        shd::applyTyped(shval, [&k2val](const auto& afr) {
+            using T = shd::applied_type_t<decltype(afr)>;
+            auto obj = afr.field.template get<T>();
+            if constexpr (std::is_same_v<T, shd::FieldType>) {
+                k2val = dtoexp::makeValueLiteral(static_cast<dto::FieldType>(to_integral(obj)));
+            } else if constexpr (std::is_same_v<T, sh::String>) {
+              k2val = dtoexp::makeValueLiteral<String>(String(std::move(obj)));
+            } else {
+              k2val = dtoexp::makeValueLiteral<T>(std::move(obj));
+            }
+        });
+    }
+    return k2val;
+}
+
+dtoexp::Expression getFilterExpression(shdexp::Expression&& shExpr) {
+    std::vector<dtoexp::Value> values;
+    values.reserve(shExpr.valueChildren.size());
+    for (auto& val: shExpr.valueChildren) {
+        values.push_back(getValue(std::move(val)));
+    }
+    std::vector<dtoexp::Expression> exprs;
+    exprs.reserve(shExpr.expressionChildren.size());
+    for (auto& cexpr: shExpr.expressionChildren) {
+        exprs.push_back(getFilterExpression(std::move(cexpr)));
+    }
+    return dtoexp::makeExpression(static_cast<dtoexp::Operation>(to_integral(shExpr.op)), std::move(values), std::move(exprs));
+}
+
 seastar::future<std::tuple<sh::Status, shd::CreateQueryResponse>>
 HTTPProxy::_handleCreateQuery(shd::CreateQueryRequest&& request) {
     K2LOG_D(log::httpproxy, "Received create query request {}", request);
-    return MakeHTTPResponse<shd::CreateQueryResponse>(sh::Statuses::S501_Not_Implemented("create query not implemented"), shd::CreateQueryResponse{});
+    auto it = _txns.find(request.timestamp);
+    if (it == _txns.end()) {
+        return MakeHTTPResponse<shd::CreateQueryResponse>(Txn_S410_Gone, shd::CreateQueryResponse{});
+    }
+    return _client.createQuery(request.collectionName, request.schemaName)
+        .then([this, req=std::move(request)] (auto&& result) mutable {
+            if(!result.status.is2xxOK()) {
+                return MakeHTTPResponse<shd::CreateQueryResponse>(sh::Status{.code = result.status.code, .message = result.status.message}, shd::CreateQueryResponse{});
+            }
+            try {
+                setQueryRecord(req.collectionName, std::move(req.key), result.query.startScanRecord);
+                setQueryRecord(req.collectionName, std::move(req.endKey), result.query.endScanRecord);
+
+                if (req.recordLimit >= 0) result.query.setLimit(req.recordLimit);
+                if (req.includeVersionMismatch) result.query.setIncludeVersionMismatch(req.includeVersionMismatch);
+                if (req.reverseDirection) result.query.setReverseDirection(req.reverseDirection);
```

Comment:
i don't think we need to check these values - just set them on the result.query

Reply:
Done